### ISSUE TITLE
Bug: proxy ignored and should be a global option

### ISSUE THREAD

**Issue by RJVB** [ferdium/ferdium-app#660]
### Avoid duplicates

- [X] I have searched the [issues tracker](https://github.com/ferdium/ferdium-app/issues) for a bug report similar to mine, in vain

### Ferdium Version

6.2.0

### What Operating System are you using?

Other Linux

### Operating System Version

custom

### What arch are you using?

x64

### Expected Behavior

1) configuring a proxy should work, e.g. when I enter the address of my Privoxy proxy I should be able to see Privoxy's settings via the URL `config.privoxy.org`.

2) there should probably be a global proxy setting (or Ferdium should use `http_proxy` and/or `https_proxy`).

### Actual Behavior

1) Proxy setting appear to be ignored (in the "Custom Website" recipe)
2) Proxy information has to be entered for each and every service which is cumbersome if you are behind a compulsory proxy

Setting `http_proxy` and `https_proxy` has no effect.

Screenshot taken with the 2 env. variables above set, plus a custom service's proxy set to the same value, and after restarting Ferdium:

![image](https://user-images.githubusercontent.com/1756884/194705816-114af512-8de1-4b79-9805-1bc3a7377721.png)

### Steps to reproduce

1. Open service settings
2. Activate proxy support and enter proxy details
3. Check if the proxy functions/is being used.

**Comment by sysarcher** [comment]
Proxy seems to work for me. I also specify the schema e.g. `http://proxy.company.com` under the `Proxy Host/IP` field and the port seperately.

**Comment by Leopere** [comment]
I disagree it should not be a global option it should be for the tab itself.  The flexibility of opting in on independent tabs is actually a pretty incredible design.

**Comment by RJVB** [comment]
On Saturday July 06 2024 06:01:20 Leopere wrote:
> The flexibility of opting in on independent tabs is actually a pretty incredible design.

The obligation to redefine it again for every service is actually also a pretty cumbersome design...

I have nothing against being able to define it at 2 levels. The global level for proxies that should be used by all services (e.g. because otherwise "no internet"), and a per-service override.

**Comment by Leopere** [comment]
Two levels is ideal but probably annoying to implement but thanks for acknowledging how you would meet me half way.  What proxy are you using I've been trying to get Privoxy working for this but like I'm not sure what to use for this goal.

**Comment by RJVB** [comment]
On Saturday July 06 2024 09:17:24 Leopere wrote:
>Two levels is ideal but probably annoying to implement but thanks for acknowledging how you would meet me half way.  What proxy are you using I've been trying to get Privoxy working for this but like I'm not sure what to use for this goal.

Privoxy is exactly what I've been using for years: it was a very useful ad blocker and privacy guard before browser extensions became commonplace. Since filing this ticket (which I forgot about...!) I mostly dropped it because I didn't want to bother setting it up to snoop my https traffic.
You'll agree with me though that using Privoxy through a global-level proxy in Ferdium would be a welcome fallback alternative for implementing extension support in Ferdium.

So yeah, Privoxy is what I'd use here. It's simple to check too if the proxy is acknowledged: just point a service to config.privoxy.org (I assume that should work too with https if properly configured).

**Comment by Leopere** [comment]
Yeah it would probably be a good fallback I'm just thinking primarily about scenarios where like for example I don't want reddit to know anything at all about me for example but I still don't mind using it from time to time.  So I would prefer to keep it at arms length and remote office workers might appreciate being able to be elsewhere.
